zokrates_core/src/proof_system/bn128/utils/libsnark.rs

```diff
@@ -14,6 +14,19 @@ fn vec_as_u8_32_array(vec: &Vec<u8>) -> [u8; 32] {
     array
 }
 
+pub fn prepare_public_inputs(public_inputs: Vec<&str>) -> (Vec<[u8; 32]>, usize) {
```

Comment:
here also I'm concerned about user errors leading to panics

Reply:
This method now accepts already parsed fields https://github.com/Zokrates/ZoKrates/pull/583/files#diff-26394101635fe9e8e900553bde8952baR17